### PR TITLE
feature: tag functions in eks(#5203)

### PR DESCRIPTION
**Pull request by mmerrill3**
Implementation for EKS functions tag_resource, untag_resource, and list_tags_for_resources

Implements #5203 

Signed-off-by: mmerrill3 <jjpaacks@gmail.com>

### CODE REVIEW

File: moto/eks/models.py [hunk 1]
```diff
@@ -631,6 +631,57 @@ def delete_nodegroup(self, cluster_name, nodegroup_name):
         cluster.nodegroup_count -= 1
         return result
 
+    def tag_resource(self, resource_arn, tags):
+        try:
```
Comment (bblommers) [inline moto/eks/models.py:635]: Can we have a docstring here to say that we only support tagging Clusters, and not managed node groups?

File: moto/eks/models.py [hunk 2]
```diff
@@ -631,6 +631,57 @@ def delete_nodegroup(self, cluster_name, nodegroup_name):
...
+                message="An error occurred (NotFoundException) when calling the UntagResource operation: Resource was not found",
+            )
+        for name in tag_keys:
+            if name in cluster.tags:
+                del cluster.tags[name]
+        return ""
+
+    def list_tags_for_resource(self, resource_arn):
+        try:
+            for x in self.clusters:
```
Comment (bblommers) [inline moto/eks/models.py:671]: Print-statements should not be in Moto

**Review by bblommers — CHANGES_REQUESTED** [review]
Looks like the linter is failing on a few things as well:
https://github.com/spulec/moto/runs/6838049781?check_suite_focus=true

File: moto/eks/models.py (hunk 1 above)
Comment (mmerrill3) [reply moto/eks/models.py:635]: yes, will update, thanks

File: moto/eks/models.py (hunk 2 above)
Comment (mmerrill3) [reply moto/eks/models.py:671]: I missed this, will update, thanks

**Comment by mmerrill3** [comment]
@bblommers forgot to install linter on VSC on new mac.  Fixed that.

**Comment by mmerrill3** [comment]
@bblommers ready for review again, thank you for your feedback.

**Review by bblommers — CHANGES_REQUESTED** [review]
Thanks for making the changes @mmerrill3. Looks like it's just the ServerMode tests that are failing now. Looking at the code, I can't immediately see why though - let me know if you want me to do some debugging on this.

**Comment by mmerrill3** [comment]
Hi @bblommers, the server tests are running in the "docker" mode, using the latest tag of the motoserver docker image.  That doesn't have the new URLs from eks/urls.py for tagging.  What normally is the process to update that docker image?  This looks like a chicken/egg problem.

**Comment by mmerrill3** [comment]
@bblommers I updated the urls to point to a specific tags function in the EKSResponse object, like other services have done (mq is one).

**Review by bblommers — APPROVED** [review]
Awesome - thanks again for all your work on this @mmerrill3!

**Pull request merged**